Source/Core/DiscIO/WIABlob.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DiscIO/WIABlob.h"
+
+#include <algorithm>
+#include <memory>
+
+#include "Common/Align.h"
+#include "Common/CommonTypes.h"
+#include "Common/File.h"
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Common/Swap.h"
+
+namespace DiscIO
+{
+WIAFileReader::WIAFileReader(File::IOFile file) : m_file(std::move(file))
+{
+  m_file.Seek(0, SEEK_SET);
```

Comment:
It seems kind of odd to put the seek into the constructor, especially given it's correcting an artifact of the current calling code that executes the constructor.

Reply:
So you mean that it would be better to put the seek inside `Create` (before the call to the constructor) instead? I was thinking it would make more sense to put it inside the constructor, since that means the constructor is making less assumptions about the calling code.